managed/services/server/server.go

```diff
@@ -510,9 +511,20 @@ func (s *Server) validateChangeSettingsRequest(ctx context.Context, req *serverp
 	if req.AlertManagerRules != "" && req.RemoveAlertManagerRules {
 		return status.Error(codes.InvalidArgument, "Both alert_manager_rules and remove_alert_manager_rules are present.")
 	}
-	if req.PmmPublicAddress != "" && req.RemovePmmPublicAddress {
-		return status.Error(codes.InvalidArgument, "Both pmm_public_address and remove_pmm_public_address are present.")
+	if req.PmmPublicAddress != "" {
+		if req.RemovePmmPublicAddress {
+			return status.Error(codes.InvalidArgument, "Both pmm_public_address and remove_pmm_public_address are present.")
+		}
+
+		publicURL, err := url.Parse(req.PmmPublicAddress)
+		if err != nil {
+			return status.Error(codes.InvalidArgument, "Provided PMM public address is not a valid URL.")
+		}
+		if publicURL.Scheme != "" {
+			return status.Error(codes.InvalidArgument, "PMM public address should not include the URL scheme")
+		}
```

Comment:
This is the new condition as I see, right? If so, what users which already saved public address with scheme before should do?

Reply:
PMM will continue to work fine for them, but Grafana domain will still be localhost.

They will need to update the public address to update Grafana domain, then we will apply this check.

Reply:
I mean user with previous PMM version was able to save public address with schema, right?

Reply:
I checked, it works in recent version: 
![image](https://github.com/percona/pmm/assets/21174514/5d290288-5fd1-4b78-b0ae-62f4045c81f3)

Reply:
I've reworked it to only use the domain part for grafana.ini